src/dsp_tools/utils/shared.py

```diff
@@ -82,20 +86,13 @@ def try_network_action(
                 print(f'{datetime.now().isoformat()}: Try reconnecting to DSP server, next attempt in {2 ** i} seconds...')
                 time.sleep(2 ** i)
                 continue
-            if hasattr(err, 'message'):
-                err_message = err.message
-            else:
-                err_message = str(err).replace('\n', ' ')
-                err_message = err_message[:150] if len(err_message) > 150 else err_message
-            raise UserError(f"{failure_msg}.\nOriginal error message for diagnostic purposes:\n{err_message}") from None
-        except Exception as exc:
-            if hasattr(exc, 'message'):
-                exc_message = exc.message
-            else:
-                exc_message = str(exc).replace('\n', ' ')
-                exc_message = exc_message[:150] if len(exc_message) > 150 else exc_message
-            raise UserError(f"{failure_msg}.\nOriginal error message for diagnostic purposes:\n{exc_message}") from None
+            logger.exception(failure_msg)
+            raise UserError(f"{failure_msg}") from None
+        except Exception:
+            logger.exception(failure_msg)
+            raise UserError(f"{failure_msg}") from None
```

Comment:
`logger.exception()` is a handy shortcut that logs the stack trace with the original error message. So the entire handling of the error message that I had done by hand until now, is not necessary any more.